### DISCUSSION TITLE
Added remote functionality

### DISCUSSION THREAD

**Discussion by mdehrlich** [GerryDazoo/Slinger#502] [Ideas]
I have 2 slingboxes, and I define 3 variations for each.   MainHD, MainLD, MainAO,, BedHD, BedLD, BedAO.  I have added a few lines of code that identify the root names (Main and Bed), and track the active config for each box . The remote sends codes to the root name, and slinger auto selects the running stream.  The remote can still work the original way with the url using the full name.  The only constraint is the the names are all the same except the last 2 characters ID the specific config. 
examples:
http://address:port/Remote/MainHD    (remote sends codes to this specific streamer - this still works ok)
http://address:port/Remote/Main   (remote lets slinger send  codes to the current streamer running on "Main")

This requires one line of code in three locations in slinger:
1) Building the dictionary when starting up.
2) Registering the current streamer when a stream is started.
3) Finding the current streamer when a remote is opened, or code received.

The motivation here is to avoid entering a new url for the remote each time a different config is used in the player.

**Comment by mdehrlich** [comment]
Note: these changes do not require current uses to make any changes to their remotes.  
Here is the added code:
Define a new global dictionary variable (i use current_stream_dict{} ). add it to all the global definitions of these functions.
1)  in function   def start_slingbox_session
after this line:   name = slinginfo.get('name', box_name).strip()
add:                 current_stream_dict[name[:-2]] =  name      # this builds the dictionary.  1 entry for each slingbox.
2) in function def connect
after these lines: 
                          if channel is None: channel = '0'
                         print('Streamer Name', streamer_name, channel)
                         if streamer_name in streamer_qs.keys():
add:                 current_stream_dict[streamer_name[:-2]] = streamer_name   # register the stream started for this slingbox
3) in function def get_streamer
after this line:  #    print('URI', text, 'Streamer', streamer, request)
add:                 if streamer in current_stream_dict: streamer = current_stream_dict[streamer] #if this is a root name, replace it with the config name of the current streamer for this slingbox.
I hope you can add this to the new release.

**Reply by pfuller88** [reply]
Hi, thanks for this.  I am interested, although I have a few questions.    Are the above 3 steps for the slingbox_server4.0.py python file?  If so,  I cannot find the "after this" code you state above in the section you name.  For example, the "name = slinginfo.get('name', box_name).strip()" has two instances in the py code, much futher down and not in the "def start_slingbox_session" function.

Hopefully I can get this to work.

**Reply by mdehrlich** [reply]
You are correct that the  "name = slinginfo.get('name', box_name).strip()" has two instances in the py code.  The new code line should be inserted after the first occurrence.   The only constraint is to have the name field in the config file as root followed by unique 2 characters.  I use roots of Main and Bed for my two boxes.  Then I attach the 2 character suffixes "HD", "LD", or "AO"  Example "MainHD" is one of my config names.    In the HTML for the remote, replace "INSERT BOXNAMES HERE&lt;/select&gt;"  with &lt;option&gt;root1&lt;/option&gt;   &lt;option&gt;root2&lt;/option&gt;  etc. followed by &lt;/select&gt;
My options would be Main and Bed.

**Reply by mdehrlich** [reply]
I may be confusing my HTML remote with the function of this change.  In my remote I can switch between the boxes without having to enter a new URL.  If you don't want to do the option thing, you can still enter the URL, but just use the root name.  No need to specify if HD, LD etc.  just /Remote/Main    The changed above will add the correct config name for the specific config that is currently running.

**Comment by mdehrlich** [comment]
I don't have release 4.  This was 3.08f. when I get 4, I'll update this
post.

**Comment by GerryDazoo** [comment]
There's been a lot of changes to that code in 4.0.  FYI here's the latest, getting very close to releasing this.

[slingbox_server_4.0.zip](https://github.com/GerryDazoo/Slinger/files/10547685/slingbox_server_4.0.zip)

I don't think the above code is needed if you use SlingerPlayer.

[SlingerPlayerFull.zip](https://github.com/GerryDazoo/Slinger/files/10547725/SlingerPlayerFull.zip)

**Reply by pfuller88** [reply]
thanks much.
